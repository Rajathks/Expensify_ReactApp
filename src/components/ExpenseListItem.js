import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({ description, id, amount, createDate }) => (
  <Link className="list_item" to={`/edit/${id}`}>
    <div >
     <h3 className="list_item__title"> {description} </h3> 
      <span className="list_item__subtitle">{moment(createDate).format(" MMMM Do - YYYY")}</span>
    </div>
    <h3 className="list_item__amount">{numeral(amount / 100).format("$0,00,000")}</h3>
  </Link>
);

export default ExpenseListItem;
