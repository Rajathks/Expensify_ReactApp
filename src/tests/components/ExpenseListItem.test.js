import ExpenseListItem from "../../components/ExpenseListItem";
import {shallow} from "enzyme";
import React from "react";
import moment from "moment";

const expenses = [
    {
        id:"1",
        description : "lGum",
        note:"sample note",
        amount : "5000",
        createDate : moment(0)
    },
    {
        id:"2",
        description : "Rent",
        note:"Rent note",
        amount : "8000",
        createDate : moment(0).subtract(4,"days").valueOf()
    },
    {
        id:"3",
        description : "Bill",
        note:"Bill note",
        amount : "9000",
        createDate : moment(0).add(5,"days").valueOf()
    }
];


test("Expense List Item Componenet Test : " , () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

