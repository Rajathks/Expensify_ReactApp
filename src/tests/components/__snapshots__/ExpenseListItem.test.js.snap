// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List Item Componenet Test :  1`] = `
<Link
  className="list_item"
  to="/edit/1"
>
  <div>
    <h3
      className="list_item__title"
    >
       
      lGum
       
    </h3>
    <span
      className="list_item__subtitle"
    >
       January 1st - 1970
    </span>
  </div>
  <h3
    className="list_item__amount"
  >
    $50
  </h3>
</Link>
`;
